controllers/chat/deleteMessage.js

```diff
@@ -0,0 +1,32 @@
+const {StreamChat} = require('stream-chat');
+
+const deleteMessage = async (req, res) => {
+  const client = new StreamChat(process.env.API_KEY, process.env.SECRET_KEY);
+  const {messageID} = req.params;
+
+  try {
+    await client.connectUser(
+      {
+        id: req.userId
+      },
+      req.token
+    );
+
+    let destroy = await client.deleteMessage(messageID);
+
+    if (!destroy) {
+      return res.status(500).json({message: 'Error deleting chat'});
+    }
+
+    return res.status(200).json({message: 'Message deleted'});
+  } catch (e) {
+    if (e.code === 4) {
+      return res.status(404).json({message: 'Message not found'});
+    }
+    return res.status(500).json({message: 'Error connecting to chat'});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

It's good that you're checking for specific error codes, but it would be more robust to also include a generic error handler for unexpected errors. This way, you can provide a more detailed error message to the client.
```diff
-    if (e.code === 4) {
-      return res.status(404).json({message: 'Message not found'});
-    }
-    return res.status(500).json({message: 'Error connecting to chat'});
+    switch(e.code) {
+      case 4:
+        return res.status(404).json({message: 'Message not found'});
+      default:
+        return res.status(500).json({message: `Unexpected error: ${e.message}`});
+    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->